pkg/server/snapshots.go

```diff
@@ -68,14 +68,25 @@ func (s *snapshotsSyncer) start() {
 // sync updates all snapshots stats.
 func (s *snapshotsSyncer) sync() error {
 	start := time.Now().UnixNano()
-	collect := func(ctx context.Context, info snapshot.Info) error {
+	var snapshots []snapshot.Info
+	// Do not call `Usage` directly in collect function, because
+	// `Usage` takes time, we don't want `Walk` to hold read lock
+	// of snapshot metadata store for too long time.
+	// TODO(random-liu): Set timeout for the following 2 contexts.
+	if err := s.snapshotter.Walk(context.Background(), func(ctx context.Context, info snapshot.Info) error {
+		snapshots = append(snapshots, info)
+		return nil
+	}); err != nil {
+		return err
```

Comment:
log here and/or maybe add context to the error?

Reply:
Will add context to the error. Thanks!